front-end/src/timers/target.jsx

```diff
@@ -0,0 +1,204 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import {ErrorFor, ShowError} from '../utils/validation_helper'
+import {Field} from 'formik'
+import classNames from 'classnames'
+import i18next from 'i18next'
+import BooleanSelect from '../ui_components/boolean_select'
+
+const Target = ({
+  type,
+  target,
+  errors,
+  touched,
+  equipment,
+  macros,
+  readOnly
+}) => {
+  const equipmentOptions = () => {
+    return equipment.map(item => {
+      return (
+        <option key={item.id} value={item.id}>
+          {item.name}
+        </option>
+      )
+    })
+  }
+
+  const macroOptions = () => {
+    return macros.map(item => {
+      return (
+        <option key={item.id} value={item.id}>
+          {item.name}
+        </option>
+      )
+    })
+  }
+
+  const buildAction = () => {
+    switch (type) {
+      case 'equipment':
+        return equipmentAction()
+      case 'reminder':
+        return reminderAction()
+      case 'macros':
+        return macroAction()
+      default:
+        return 'Unknown type:' + type
+    }
+  }
+
+  const macroAction = () => {
+    return (
+      <React.Fragment>
+        <div className='col-12 col-sm-4 col-lg-3 order-lg-6 col-xl-2'>
+          <div className='form-group'>
+            <label htmlFor='nsme'>{i18next.t('timers:macro')}</label>
+            <Field
+              name='name'
+              component='select'
+              disabled={readOnly}
+              className={classNames('custom-select', {
+                'is-invalid': ShowError('name', touched, errors)
+              })}
+            >
+              <option value='' className='d-none'>-- {i18next.t('select')} --</option>
+              {macroOptions()}
```

Comment:
Why not putting the empty option directly into macroOptions()

Reply:
I didn't get what you mean. The macros list is coming from props

Reply:
You map it in `macroOptions`, which takes macros from props. So i'd personnaly adf the empty field in macroOptions instead of the jsx